test/unit/services/test_image.py

```diff
@@ -247,3 +247,33 @@ def test_list(self):
         self.mock_list()
         self.call_list(entity_id=str(ObjectId()), primary=False)
         self.check_list_success()
+
+
+class DeleteDSL(ImageServiceDSL):
+    """Base class for `delete` tests."""
+
+    _delete_image_id: str
+
+    def call_delete(self, image_id: str) -> None:
+        """
+        Calls the `ImageService` `delete` method.
+
+        :param image_id: ID of the image to be deleted.
+        """
+
+        self._delete_image_id = image_id
+        self.image_service.delete(image_id)
+
+    def check_delete_success(self) -> None:
+        """Checks that a prior call to `call_delete` worked as expected."""
+
+        self.mock_image_repository.delete.assert_called_once_with(self._delete_image_id)
```

Comment:
Shouldn't this also assert the store delete was called?

Reply:
It should be, thank you for pointing it out.